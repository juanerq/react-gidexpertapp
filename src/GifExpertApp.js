import { useState } from "react"
import AddCategory from "./components/AddCategory"
import { GifGrid } from "./components/GifGrid"

const GifExpertApp = ({ defaultCategories = [] }) => {

  const [categories, setCategory] = useState(defaultCategories)


  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategory={ setCategory } />
      {/* <button onClick={ handleAdd }>Add Category</button> */}
      <hr />
      <ol>
        {
          categories.map(category => (
            <GifGrid 
              key={ category } 
              category={ category }
            />
          ))
        }
      </ol>
    </>
  )
}

export default GifExpertApp