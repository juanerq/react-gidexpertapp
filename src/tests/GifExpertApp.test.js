import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Component <GifExpertApp/>', () => {
    
    test('It should display GifExpertApp Item correctly', () => {
        const wrapper = shallow( <GifExpertApp/> )
        expect(wrapper).toMatchSnapshot()
    })

    test('Should display a list of category', () => {
        const categories = ['Attack on tittan', 'Demon slayer']
        const wrapper = shallow( <GifExpertApp defaultCategories={categories}/> )

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
})