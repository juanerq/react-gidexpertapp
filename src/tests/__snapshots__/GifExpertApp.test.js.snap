// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <GifExpertApp/> It should display GifExpertApp Item correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Component <GifExpertApp/> Should display a list of category 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Attack on tittan"
      key="Attack on tittan"
    />
    <GifGrid
      category="Demon slayer"
      key="Demon slayer"
    />
  </ol>
</Fragment>
`;
